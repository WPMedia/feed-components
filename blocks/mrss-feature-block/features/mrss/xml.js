import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'
import { findVideo } from '@wpmedia/feeds-find-video-stream'

const jmespath = require('jmespath')

const rssTemplate = (
  elements,
  {
    channelTitle,
    channelDescription,
    channelPath,
    channelCopyright,
    channelTTL,
    channelUpdatePeriod,
    channelUpdateFrequency,
    channelCategory,
    channelLogo,
    imageTitle,
    imageCaption,
    imageCredits,
    itemTitle,
    itemDescription,
    pubDate,
    itemCredits,
    itemCategory,
    includeContent,
    resizerURL,
    domain,
    feedTitle,
    feedLanguage,
    videoInfo,
    mrssBuildContent,
  },
) => ({
  rss: {
    '@xmlns:atom': 'http://www.w3.org/2005/Atom',
    '@version': '2.0',
    '@xmlns:media': 'http://search.yahoo.com/mrss/',
    channel: {
      title: `${channelTitle || feedTitle + ' Videos'}`,
      link: domain,
      'atom:link': {
        '@href': `${domain}${channelPath}`,
        '@rel': 'self',
        '@type': 'application/rss+xml',
      },
      description: channelDescription || feedTitle,
      pubDate: moment.utc(Date.now()).format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: `${channelTitle || feedTitle + ' Videos'}`,
          link: domain,
        },
      }),

      item: elements.map((s) => {
        const url = `${domain}${s.website_url || s.canonical_url}`
        const img =
          s.promo_items && (s.promo_items.basic || s.promo_items.lead_art)
        const videoStream = findVideo(s, videoInfo)

        return {
          title: `${jmespath.search(s, itemTitle)}`,
          link: url,
          ...(itemCredits &&
            (jmespath.search(img, itemCredits) || []).length && {
              'media:credit': {
                '@role': 'producer',
                $: jmespath.search(img, itemCredits).join(','),
              },
            }),
          ...(itemDescription && {
            description: jmespath.search(s, itemDescription),
          }),
          guid: {
            '@isPermaLink': false,
            '#': url,
          },
          referenceid: s._id,
          pubDate: moment
            .utc(s[pubDate])
            .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),

          'media:content': {
            '@': {
              isDefault: 'true',
              ...(s.duration && {
                duration: Math.trunc(s.duration / 1000),
              }),
              ...(videoStream && {
                ...(videoInfo.url && { url: videoInfo.url }),
                ...(videoInfo.height && { height: videoInfo.height }),
                ...(videoInfo.width && { width: videoInfo.width }),
                ...(videoInfo.bitrate && { bitrate: videoInfo.bitrate }),
                ...(videoInfo.stream_type && {
                  type: videoInfo.stream_type,
                  '#': videoStream,
                }),
              }),
            },
          },
          'media:keywords': (
            jmespath.search(s, 'taxonomy.seo_keywords[*]') || []
          ).join(','),
          ...(s.description ||
            (s.subheadlines &&
              s.subheadlines.basic && {
                'media:caption': {
                  $: s.description || s.subheadlines.basic,
                },
              })),
          'media:transcript': s.transcript,
          ...(s.taxonomy &&
            s.taxonomy.primary_section &&
            s.taxonomy.primary_section.name && {
              'media:category': s.taxonomy.primary_section.name,
            }),
          ...(img &&
            img.url && {
              'media:thumbnail': {
                '@url': buildResizerURL(img.url, resizerKey, resizerURL),
              },
            }),
        }
      }),
    },
  },
})

export function Mrss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  function MrssBuildContent() {
    BuildContent.call(this)
  }
  const mrssBuildContent = new MrssBuildContent()

  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    mrssBuildContent,
  })
}

Mrss.propTypes = {
  customFields: PropTypes.shape({
    channelPath: PropTypes.string.tag({
      label: 'Path',
      group: 'Channel',
      description:
        'Path to the feed, excluding the domain, defaults to /arcio/mrss',
      defaultValue: '/arcio/mrss/',
    }),
    selectVideo: PropTypes.kvp.tag({
      label: 'Select video using',
      group: 'Video',
      description:
        'This criteria is used to filter videos encoded in the streams array',
      defaultValue: {
        height: '',
        width: '',
        bitrate: 5400,
        stream_type: 'mp4',
      },
    }),
    ...generatePropsForFeed('rss', PropTypes),
  }),
}
Mrss.label = 'MRSS'
export default Consumer(Mrss)
