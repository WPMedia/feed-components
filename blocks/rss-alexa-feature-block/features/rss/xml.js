import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'

const jmespath = require('jmespath')

const cheerio = require('cheerio')

const getURLType = (url) => {
  const imgRegex = new RegExp(/(jpe?g|png|webp|PNG|JPG|WEBP)$/);
  const audioRegex = new RegExp('^(https?|ftp|file):\/\/(www.)?(.*?)\.(mp3)$');
  const videoRegex = new RegExp('^(https?|ftp|file):\/\/(www.)?(.*?)\.(mp4|ts)$');
  const uri = url.split(".");
  const type = uri[uri.length -1];
  const defaultType =  {label: 'Audio',type: `audio/${type}`};
  if(imgRegex.test(url)) {
    return {label: 'image',type: `image/${type}`};
  }
  if(audioRegex.test(url)){
    return {label: 'Audio',type: `audio/${type}`};
  }
  if(videoRegex.test(url)){
    return {label: 'Video',type: `video/${type}`};
  }
  
  return defaultType;
}

const rssTemplate = (
  elements,
  {
    audioAvailable,
    channelTitle,
    channelDescription,
    channelCopyright,
    channelTTL,
    channelCategory,
    channelLogo,
    itemTitle,
    pubDate,
    itemCategory,
    includeContent,
    resizerURL,
    resizerWidth,
    resizerHeight,
    domain,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  },
) => ({
  rss: {
    '@version': '2.0',
    channel: {
      ...(channelTitle && { title: channelTitle }),
      link: `${domain}`,
      ...(channelDescription && { description: channelDescription }),
      lastBuildDate: moment
        .utc(new Date())
        .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(feedLanguage && { language: feedLanguage }),
      ...(channelCategory && { category: channelCategory }),
      ...(channelCopyright && {
        copyright: channelCopyright,
      }), // TODO Add default logic
      ...(channelTTL && { ttl: channelTTL }),

      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: channelTitle || feedTitle,
          link: domain,
        },
      }),

      item: elements.map((s) => {
        let body, category,
        enclosureurl = jmespath.search(s, audioAvailable);
        const url = `${domain}${s.website_url || s.canonical_url || ''}`
        return {
          title: jmespath.search(s, itemTitle) || '',
          link: url,
          guid: s._id,
          pubDate: s[pubDate],
          ...(enclosureurl  &&
          {
            enclosure: {
              '@url': enclosureurl,
              '@label':  getURLType(enclosureurl).label,
              '@group': getURLType(enclosureurl).label,
              '@type': getURLType(enclosureurl).type
            }
          }),
          ...(itemCategory &&
            (category = jmespath.search(s, itemCategory)) &&
            category && { category: category }),
            description: cheerio
            .load(
              rssBuildContent.parse(
                s.content_elements || [],
                includeContent,
                domain,
                resizerKey,
                resizerURL,
                resizerWidth,
                resizerHeight,
              ) || '',
            )
            .text()
        }
      }),
    },
  },
})

export function Rss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  const { width = 0, height = 0 } = customFields.resizerKVP || {};
  

  const rssBuildContent = new BuildContent()
 
  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    resizerWidth: width,
    resizerHeight: height,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  })
}

Rss.propTypes = {
  customFields: PropTypes.shape({
    audioAvailable: PropTypes.string.tag({
      label: 'Audio',
      group: 'Audio',
      description: 'description',
      defaultValue: `content_elements[?type=='audio'].streams[].url|[0]`,
    }),
    ...generatePropsForFeed('rss', PropTypes, ['featuredImage', 'includePromo']),
  }),
}
Rss.label = 'RSS Alexa'
export default Consumer(Rss)
