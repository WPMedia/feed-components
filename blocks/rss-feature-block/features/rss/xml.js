import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'

const jmespath = require('jmespath')

const rssTemplate = (
  elements,
  {
    channelTitle,
    channelDescription,
    channelPath,
    channelCopyright,
    channelTTL,
    channelUpdatePeriod,
    channelUpdateFrequency,
    channelCategory,
    channelLogo,
    imageTitle,
    imageCaption,
    imageCredits,
    itemTitle,
    itemDescription,
    pubDate,
    itemCategory,
    includePromo,
    includeContent,
    resizerURL,
    domain,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  },
) => ({
  rss: {
    '@xmlns:atom': 'http://www.w3.org/2005/Atom',
    '@xmlns:content': 'http://purl.org/rss/1.0/modules/content/',
    '@xmlns:dc': 'http://purl.org/dc/elements/1.1/',
    '@xmlns:sy': 'http://purl.org/rss/1.0/modules/syndication/',
    '@version': '2.0',
    ...(includePromo && {
      '@xmlns:media': 'http://search.yahoo.com/mrss/',
    }),
    channel: {
      title: `${channelTitle || feedTitle}`,
      link: `${domain}`,
      'atom:link': {
        '@href': `${domain}${channelPath}`,
        '@rel': 'self',
        '@type': 'application/rss+xml',
      },
      description: `${channelDescription || feedTitle + ' News Feed'}`,
      lastBuildDate: moment
        .utc(new Date())
        .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(feedLanguage && { language: feedLanguage }),
      ...(channelCategory && { category: channelCategory }),
      ...(channelCopyright && {
        copyright: channelCopyright,
      }), // TODO Add default logic
      ...(channelTTL && { ttl: channelTTL }),
      ...(channelUpdatePeriod && {
        'sy:updatePeriod': channelUpdatePeriod,
      }),
      ...(channelUpdateFrequency && {
        'sy:updateFrequency': channelUpdateFrequency,
      }),
      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: `${channelTitle || feedTitle}`,
          link: `${domain}`,
        },
      }),

      item: elements.map((s) => {
        let author, body, category
        const url = `${domain}${s.website_url || s.canonical_url}`
        const img =
          s.promo_items && (s.promo_items.basic || s.promo_items.lead_art)
        return {
          title: `${jmespath.search(s, itemTitle)}`,
          link: url,
          guid: {
            '#': url,
            '@isPermaLink': true,
          },
          ...((author = jmespath.search(s, 'credits.by[].name')) &&
            author && {
              'dc:creator': author.join(','),
            }),
          description: { $: jmespath.search(s, itemDescription) },
          pubDate: moment
            .utc(s[pubDate])
            .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
          ...(itemCategory &&
            (category = jmespath.search(s, itemCategory)) &&
            category && { category: category }),
          ...(includeContent !== '0' &&
            (body = rssBuildContent.parse(
              s.content_elements,
              includeContent,
              domain,
              resizerKey,
              resizerURL,
            )) &&
            body && {
              'content:encoded': {
                $: body,
              },
            }),
          ...(includePromo &&
            img &&
            img.url && {
              'media:content': {
                '@type': 'image/jpeg',
                '@url': buildResizerURL(img.url, resizerKey, resizerURL),
                ...(jmespath.search(img, imageCaption) && {
                  'media:description': {
                    '@type': 'plain',
                    $: jmespath.search(img, imageCaption),
                  },
                }),
                ...(jmespath.search(img, imageTitle) && {
                  'media:title': {
                    $: jmespath.search(img, imageTitle),
                  },
                }),
                ...((jmespath.search(img, imageCredits) || []).length && {
                  'media:credit': {
                    '@role': 'author',
                    '@scheme': 'urn:ebu',
                    '#': jmespath.search(img, imageCredits).join(','),
                  },
                }),
              },
            }),
        }
      }),
    },
  },
})

export function Rss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  const rssBuildContent = new BuildContent()

  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  })
}

Rss.propTypes = {
  customFields: PropTypes.shape({
    channelPath: PropTypes.string.tag({
      label: 'Path',
      group: 'Channel',
      description:
        'Path to the feed, excluding the domain, defaults to /arcio/rss',
      defaultValue: '/arcio/rss/',
    }),
    ...generatePropsForFeed('rss', PropTypes),
  }),
}
Rss.label = 'Standard RSS'
export default Consumer(Rss)
