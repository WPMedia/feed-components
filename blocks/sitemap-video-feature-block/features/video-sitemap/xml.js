import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { buildResizerURL } from '@wpmedia/feeds-resizer'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { findVideo } from '@wpmedia/feeds-find-video-stream'
import jmespath from 'jmespath'

const sitemapTemplate = (
  elements,
  {
    lastMod,
    promoItemsJmespath = 'promo_items.basic || promo_items.lead_art',
    videoDescription,
    sitemapVideoSelect,
    videoTitle,
    domain,
    resizerURL,
    resizerWidth,
    resizerHeight,
  },
) => ({
  urlset: {
    '@xmlns': 'http://www.sitemaps.org/schemas/sitemap/0.9',
    '@xmlns:video': 'http://www.google.com/schemas/sitemap-video/1.1',
    url: elements.map((v) => {
      let title, description, category
      // Get promo image object to determine thumbnail_loc
      let img = promoItemsJmespath && jmespath.search(v, promoItemsJmespath)
      if (img && !img.url && img.promo_image) img = img.promo_image // video
      // Find content_loc based on searchObject site property.
      const contentLoc = findVideo(v, sitemapVideoSelect)

      return {
        loc: `${domain}${v.website_url || v.canonical_url}`,
        'video:video': {
          ...(contentLoc && { 'video:content_loc': contentLoc.url }),
          ...(v.duration && {
            'video:duration': Math.trunc(v.duration / 1000),
          }),
          ...(img &&
            img.url && {
              'video:thumbnail_loc': buildResizerURL(
                img.url,
                resizerKey,
                resizerURL,
                resizerWidth,
                resizerHeight,
                img,
              ),
            }),
          ...(videoTitle &&
            (title = jmespath.search(v, videoTitle)) &&
            title && {
              'video:title': { $: title },
            }),
          ...(videoDescription &&
            (description = jmespath.search(v, videoDescription)) &&
            description && {
              'video:description': {
                $: description,
              },
            }),
          ...((category = jmespath.search(
            v,
            'taxonomy.primary_section.name',
          )) &&
            category && {
              'video:category': category,
            }),
        },
        ...{ lastmod: v[lastMod] },
      }
    }),
  },
})

export function VideoSitemap({ globalContent, customFields, arcSite }) {
  const { resizerURL = '', feedDomainURL = '' } = getProperties(arcSite)
  const { width = 0, height = 0 } = customFields.resizerKVP || {}

  // can't return null for xml return type, must return valid xml template
  return sitemapTemplate(globalContent.content_elements || [], {
    ...customFields,
    domain: feedDomainURL,
    resizerURL,
    resizerWidth: width,
    resizerHeight: height,
  })
}

VideoSitemap.propTypes = {
  customFields: PropTypes.shape({
    videoTitle: PropTypes.string.tag({
      label: 'Video Title',
      group: 'Format',
      description:
        'Which ANS field should be used for the title.  Defaults to headlines.basic',
      defaultValue: 'headlines.basic',
    }),
    videoDescription: PropTypes.string.tag({
      label: 'Description',
      group: 'Format',
      description:
        'Which ANS field should be used for the description. Defaults to description.basic || subheadlines.basic',
      defaultValue: 'description.basic || subheadlines.basic',
    }),
    sitemapVideoSelect: PropTypes.kvp.tag({
      label: 'Video Encoding',
      group: 'Format',
      description: 'Which criteria should be used to filter video encodings',
      defaultValue: { bitrate: 5400, stream_type: 'mp4' },
    }),
    ...generatePropsForFeed('sitemap', PropTypes, ['videoSelect']),
  }),
}
VideoSitemap.label = 'Sitemap Video'
VideoSitemap.icon = 'browser-page-hierarchy'
export default Consumer(VideoSitemap)
