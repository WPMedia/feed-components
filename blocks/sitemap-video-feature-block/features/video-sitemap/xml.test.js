// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
import { VideoSitemap } from './xml'

const globalContent = {
  content_elements: [
    {
      type: 'video',
      version: '0.8.0',
      canonical_url:
        '/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/',
      canonical_website: 'demo',
      short_url:
        '/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/',
      created_date: '2020-04-16T19:40:20Z',
      last_updated_date: '2020-04-16T19:55:14Z',
      publish_date: '2020-04-16T19:55:25Z',
      first_publish_date: '2020-04-16T19:55:25Z',
      display_date: '2020-04-16T19:55:25Z',
      headlines: {
        basic: 'Inexact Odyssey, A Volcom Snowboarding Film',
        title: 'Ababe beso bela',
      },
      subheadlines: {
        basic: 'Video subheadline',
      },
      description: {
        basic: 'Video Description',
      },
      credits: {},
      taxonomy: {
        tags: [],
        primary_section: {
          _id: '/video',
          name: 'Video',
          description: 'Latest videos from around the site.',
        },

        seo_keywords: ['sample', 'demo'],
      },
      promo_items: {
        basic: {
          type: 'image',
          version: '0.5.8',
          credits: {},
          caption: 'Inexact Odyssey',
          url: 'https://dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg',
          width: 1440,
          height: 1080,
        },
      },
      duration: 789060,
      video_type: 'clip',
      streams: [
        {
          height: 1080,
          width: 1440,
          filesize: 474649052,
          stream_type: 'ts',
          url: 'https://d1tf6ure8fkb0l.cloudfront.net/wp-demo/2020/04/16/5e98b4a44cedfd00019ccd14/t_45faafc2013a4185b449ceb539e3df71_name_Inexact_Odyssey__A_Volcom_Snowboarding_Film/hd.m3u8',
          bitrate: 4500,
          provider: 'mediaconvert',
        },
        {
          height: 1080,
          width: 1440,
          filesize: 549824236,
          stream_type: 'ts',
          url: 'https://d1tf6ure8fkb0l.cloudfront.net/wp-demo/2020/04/16/5e98b4a44cedfd00019ccd14/t_45faafc2013a4185b449ceb539e3df71_name_Inexact_Odyssey__A_Volcom_Snowboarding_Film/hlsv4_master.m3u8',
          bitrate: 5400,
          provider: 'mediaconvert',
        },
        {
          height: 1080,
          width: 1440,
          filesize: 545772131,
          stream_type: 'mp4',
          url: 'https://d1tf6ure8fkb0l.cloudfront.net/wp-demo/2020/04/16/5e98b4a44cedfd00019ccd14/t_45faafc2013a4185b449ceb539e3df71_name_Inexact_Odyssey__A_Volcom_Snowboarding_Film/file_1920x1080-5400-v4.mp4',
          bitrate: 5400,
          provider: 'mediaconvert',
        },
      ],
      subtitles: {},
      promo_image: {
        type: 'image',
        version: '0.5.8',
        credits: {},
        caption: 'Inexact Odyssey',
        url: 'https://dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg',
        width: 1440,
        height: 1080,
      },
      website: 'demo',
      website_url:
        '/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/',
    },
    {
      type: 'video',
      version: '0.8.0',
      canonical_url: '/video/2021/04/16/kittend/',
      last_updated_date: '2021-04-16T19:55:14Z',
      display_date: '2021-04-16T19:55:25Z',
      headlines: {
        basic: 'Inexact Odyssey, A Volcom Snowboarding Film',
        title: 'Ababe beso bela',
      },
      subheadlines: {
        basic: 'Video subheadline',
      },
      description: {
        basic: 'Video Description',
      },
      credits: {},
      taxonomy: {
        tags: [],
        primary_section: {
          _id: '/video',
          name: 'Video',
          description: 'Latest videos from around the site.',
        },

        seo_keywords: ['sample', 'demo'],
      },
      promo_items: {
        mobile: {
          type: 'image',
          version: '0.5.8',
          credits: {},
          caption: 'Inexact Odyssey',
          url: 'https://dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg',
          width: 1440,
          height: 1080,
        },
      },
      duration: 789060,
      video_type: 'clip',
      streams: [
        {
          height: 1080,
          width: 1440,
          filesize: 545772131,
          stream_type: 'mp4',
          url: 'https://d1tf6ure8fkb0l.cloudfront.net/wp-demo/2020/04/16/5e98b4a44cedfd00019ccd14/t_45faafc2013a4185b449ceb539e3df71_name_Inexact_Odyssey__A_Volcom_Snowboarding_Film/file_1920x1080-5400-v4.mp4',
          bitrate: 5400,
          provider: 'mediaconvert',
        },
      ],
      website: 'demo',
      website_url:
        '/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/',
    },
  ],
}

it('returns template with default values', () => {
  const videoSitemap = VideoSitemap({
    arcSite: 'the-globe',
    globalContent: globalContent,
    customFields: {
      lastMod: 'last_updated_date',
      videoTitle: 'headlines.basic',
      sitemapVideoSelect: { bitrate: 5400, stream_type: 'mp4' },
      videoDescription: 'description.basic || subheadlines.basic',
    },
  })
  expect(videoSitemap).toMatchSnapshot()
})

it('returns template with headlines.title, description.basic 2000 values', () => {
  const videoSitemap = VideoSitemap({
    arcSite: 'the-globe',
    globalContent: globalContent,
    customFields: {
      lastMod: 'last_updated_date',
      videoTitle: 'headlines.title',
      sitemapVideoSelect: { bitrate: 5400, stream_type: 'ts' },
      videoDescription: 'description.basic',
    },
  })
  expect(videoSitemap).toMatchSnapshot()
})
it('returns template with subheadlines.basic values', () => {
  const videoSitemap = VideoSitemap({
    arcSite: 'the-globe',
    globalContent: globalContent,
    customFields: {
      lastMod: 'display_date',
      videoTitle: '',
      sitemapVideoSelect: { foo: 'bar' },
      videoDescription: '',
    },
  })
  expect(videoSitemap).toMatchSnapshot()
})
