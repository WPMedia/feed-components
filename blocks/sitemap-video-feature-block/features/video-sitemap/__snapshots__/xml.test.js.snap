// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns template with default values 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "@xmlns:video": "http://www.google.com/schemas/sitemap-video/1.1",
    "url": Array [
      Object {
        "lastmod": "2020-04-16T19:55:14Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:description": Object {
            "$": "Video Description",
          },
          "video:duration": 789,
          "video:thumbnail_loc": "hi/abcdefghijklmnopqrstuvwxyz=/dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg",
          "video:title": Object {
            "$": "Inexact Odyssey, A Volcom Snowboarding Film",
          },
        },
      },
      Object {
        "lastmod": "2021-04-16T19:55:14Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:description": Object {
            "$": "Video Description",
          },
          "video:duration": 789,
          "video:title": Object {
            "$": "Inexact Odyssey, A Volcom Snowboarding Film",
          },
        },
      },
    ],
  },
}
`;

exports[`returns template with empty values 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "@xmlns:video": "http://www.google.com/schemas/sitemap-video/1.1",
    "url": Array [
      Object {
        "lastmod": undefined,
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:duration": 789,
        },
      },
      Object {
        "lastmod": undefined,
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:duration": 789,
        },
      },
    ],
  },
}
`;

exports[`returns template with headlines.title, description.basic 2000 values 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "@xmlns:video": "http://www.google.com/schemas/sitemap-video/1.1",
    "url": Array [
      Object {
        "lastmod": "2020-04-16T19:55:14Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:description": Object {
            "$": "Video Description",
          },
          "video:duration": 789,
          "video:thumbnail_loc": "hi/abcdefghijklmnopqrstuvwxyz=/dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg",
          "video:title": Object {
            "$": "Ababe beso bela",
          },
        },
      },
      Object {
        "lastmod": "2021-04-16T19:55:14Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:description": Object {
            "$": "Video Description",
          },
          "video:duration": 789,
          "video:title": Object {
            "$": "Ababe beso bela",
          },
        },
      },
    ],
  },
}
`;

exports[`returns template with subheadlines.basic values 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "@xmlns:video": "http://www.google.com/schemas/sitemap-video/1.1",
    "url": Array [
      Object {
        "lastmod": "2020-04-16T19:55:25Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:duration": 789,
          "video:thumbnail_loc": "hi/abcdefghijklmnopqrstuvwxyz=/dv8csq7v0ltdn.cloudfront.net/04-16-2020/t_95d8de29ea3b41caac132f0462c5c71a_name_file_1920x1080_5400_v4_.jpg",
        },
      },
      Object {
        "lastmod": "2021-04-16T19:55:25Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/video/2020/04/16/inexact-odyssey-a-volcom-snowboarding-film/",
        "video:video": Object {
          "video:category": "Video",
          "video:content_loc": "https://d3ujdjwa458jgt.cloudfront.net/out/v1/87998c783fb94bf0b965847d5c8b4392/index.m3u8",
          "video:duration": 789,
        },
      },
    ],
  },
}
`;
